### ISSUE TITLE
False positive for no-empty-first-line check with function that returns an anonymous implementation

### ISSUE THREAD

**Issue by lwasyl** [pinterest/ktlint#647]
For the following code:
```kotlin
interface Foo {

    fun foo()
}

fun fooBuilder() = object : Foo {

    override fun foo() {
        TODO()
    }
}
```
KtLint complains for the empty line after `Foo {` and before `override` with `experimental:no-empty-first-line-in-method-block] First line in a method block should not be empty`. This is a false positive, since that empty line is not part of the function declaration, but of a class declaration.

**Comment by idntfy** [comment]
What can be used now to ensure there is no empty line after `Foo {`?

**Comment by Tapchicoma** [comment]
@idntfy most probably it is required new rule, please open a separate issue

**Comment by lwasyl** [comment]
@idntfy Do you want to treat anonymous classes differently than regular classes?

**Comment by idntfy** [comment]
I want everything to be treated the same way, for me it seems this issue proposes to treat it differently by allowing empty line for anonymous classes while not allowing it in other places
